### DISCUSSION TITLE
[Projects Beta] Card appearing with "you cant see this item"

### DISCUSSION THREAD

**Discussion by ScottGuymer** [community/community#8550] [Projects and Issues]
When trying to add an issue from a repository with internal visibility I end up with a card stuck on the board I can't interact with.

![image](https://user-images.githubusercontent.com/956825/145227467-96d0dea5-a542-46f6-80d4-17611811800c.png)

This specific card is PR on an internal repo. A PR that I created but only have default internal level permissions to the repo. I can see the repo and the PR but can't change labels etc

Once added it appears in a "No Status" column that was not there before and I'm unable to do anything with this card, not even delete it. For users who have more permissions on the repo it appears in the Todo column, and they can interact with it as expected

At the very minimum it should show the basic details of the PR that I can already see, show it in the same column that it appears in for the user with the rights. But should not let me move or interact with it on the board.

**Comment by ghost** [comment]
This is annoying. It takes one full column and I can't find what this issue is. Other members of the team cannot see this issue at all

**Reply by ScottGuymer** [reply]
It seems to be based on your level of access to the repo that the issue resides in. Even if in your org you have default view access to all internal repos.

You are right its quite annoying and even though it doesn't display on the project board the user can usually see the issue if you send them the link. It doesn't match up the permissions from the board and the visibility from the repo.

**Reply by aholmes** [reply]
That's exactly it. I had a user seeing the same thing. Once I added their team to the individual repositories, it went away. This was presenting across two org projects and three repositories, so I was able to repeat the exercise and verify.

**Comment by szymi-** [comment]
I have the same problem. I am unable to remove the issue that I added previously to the beta board. I cannot see any details about it. I am the creator of the issue, I can see it perfectly well when accessing the link from the repo. I am not one of the repo admins and have no need to be. We want to have a board for many teams, many repos, people with different access rights should be able to collaborate using the board.

**Comment by dustinsgoodman** [comment]
I think I'm having the same issue as described here but might have some reproduction steps. I have a project board set up in a client's org. My team is added to the projects directly instead of the organization as a team. I'm able to see and add issues to the project board but then they go private after a second. I think not being an org member causes this problem. Is there a workaround here?

**Reply by dustinsgoodman** [reply]
Discovered another wrinkle to add to the above description - the org that I'm having difficulties with is an enterprise org and the repos that I'm having difficulty viewing are "Internal" repos. Hope that helps in debugging!

Confirmed that this is not an issue if the repo is private and a member is not a part of the org.

**Reply by ScottGuymer** [reply]
Yes we are running on an org where all repos are internal by default

**Reply by dustinsgoodman** [reply]
I ran a test with an admin on the other org where I brought him into my org which doesn't have internal projects and found that he had 0 issues. I bet there's some line of code on the permissions like:
```
user.hasPermissionToView([REPO:PUBLIC, REPO:PRIVATE])
```
and they just forgot the internal type.

**Comment by pawelru** [comment]
How about make it configurable to enable/disable showing this card?
The use case I am facing is two teams working on the same public board but at the same time they want to add to that board issues from their private repos. "you can't see this items" just looks bad and pollutes the views.

**Comment by pawelru** [comment]
I think I have found a workaround that would work as long as you don't use cards. Please add a `is:issue` filter on the view (worked for both table and board). This thing is being recognised as a card which you can filter-out.

**Reply by dustinsgoodman** [reply]
Attempted this work around just now, it just filters out everything. 😢

**Reply by pawelru** [reply]
Hmmm... Please have at look at my playground where I tested this: https://github.com/users/pawelru/projects/2/views/2
There are two repos `test_01` which is open and `test_02` which is is private

This is what I am seeing (full access):
![image](https://user-images.githubusercontent.com/12943682/168607759-1ff4068e-63c3-4e92-93ba-e8dbd8c20a74.png)

This is what you should see (taken from incognito mode)
<img width="1096" alt="image" src="https://user-images.githubusercontent.com/12943682/168607882-0c2cf903-cd51-465c-9f0b-50a0892eb7f3.png">

Without a filter:
<img width="1453" alt="image" src="https://user-images.githubusercontent.com/12943682/168607978-fd9d2e72-c2f8-41b1-9cd0-20835ba706f6.png">

**Reply by dustinsgoodman** [reply]
Ah I see - it removes the unusable items from the board. This isn't the crux of the issue though. The real issue is the visibility on those items isn't being set correctly. Other examples here are people have full access to the issue and its repo as well full access to the project board but because they're not a part of the org, the card isn't displaying correctly. Or at least, that's my key issue.

**Comment by mylaluna** [comment]
I have a test private project created two months ago. I am the only user of this project as well.
Today, I just realised I cannot see many cards which I created myself. 
This looks like a permission bug to me. Nobody can access the content of these cards now. 
I think have no other solution but to terminate the entire project.

**Comment by hugobrancowb** [comment]
I found this bug after I deleted the parent repository of the issue created. It's not an access problem because the repo is gone now

**Comment by fkarg** [comment]
This problem continues to persist in mid-2023. Yes, adding `is:issue` works, but it's a workaround, not a solution.

**Reply by vps-victor** [reply]
It worked fine for me, thanks.
Since I use all types of cards, I added the remaining values:
​`is:issue,closed,open,merged,pr`
